python/mqt/qmap/ph/unitary_to_phase_compilation.py

```
@@ -0,0 +1,615 @@
+            bottom_mode = top_mode + 1
+
+            if layer >= start_mzi_layer and top_mode in target_modes and bottom_mode in target_modes:
+                computation_bs_indices.extend((current_bs_idx, current_bs_idx + 1))
+
+            current_bs_idx += 2
+
+    indices_tensor = torch.tensor(computation_bs_indices, dtype=torch.long)
+    return all_bs_values[indices_tensor], indices_tensor
+
+
+def optimize_unitary_subcircuit_parameters(
```

Ruff (0.15.21)

[warning] Too many branches (13 > 12)

(PLR0912)